premailer/premailer.py

```diff
@@ -132,33 +167,64 @@ def __init__(self, html, base_url=None,
                  disable_basic_attributes=None,
                  disable_validation=False,
                  cache_css_parsing=True,
+                 cache_css_parsing_size=1000,
+                 cache_css_output=True,
+                 cache_css_output_size=1000,
```

Comment:
I don't think these need to be here at all. They're operational, not functional. They clutter up the code and obscures the options. 

The likelihood of someone preferring to only set it to 500 or 10_000 is extremely rare and I think there are much better ways to do that. Like this:

```python
import os

def function_cache():
    ...snip...
    # indicator of cache missed
    sentinel = object()

    max_cache_entries = int(os.environ.get('PREMAILER_MAX_CACHE_ENTRIES', 1000))

    @functools.wraps(func)
    def inner(*args, **kwargs):
        ...
    return inner
```

That will make the code a *lot* simpler. It'll give all the power to your use case too